### PR TITLE
chore: update OTel metrics to v0.35.0

### PR DESCRIPTION
**Pull request by vgarvardt**
Refs https://github.com/open-telemetry/opentelemetry-go/releases/tag/v1.12.0

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,7 +49,8 @@ Compatibility
 
 |         `otelsql`          | `go.opentelemetry.io/otel/trace` | `go.opentelemetry.io/otel/sdk/metric` |
 |:--------------------------:|:--------------------------------:|:-------------------------------------:|
-| `v0.7.*` <sup>&nbsp;</sup> |      `v1.11.1` ~> `latest`       |        `v0.33.0` ~> `latest`          |
+| `v0.8.*` <sup>&nbsp;</sup> |      `v1.12.0` ~> `latest`       |         `v0.35.0` ~> `latest`         |
+| `v0.7.*` <sup>&nbsp;</sup> |      `v1.11.1` ~> `latest`       |         `v0.33.0` ~> `latest`         |
```
Comment (nhatthm) [inline README.md]:
```suggestion
| `v0.7.*` <sup>&nbsp;</sup> |      `v1.11.1` ~> `latest`       |        `v0.33.0` ~> `v0.34.x`         |
```

**Comment by nhatthm** [comment]
I thought I was late 😂 now I see it's only 4hrs 🙈

**Comment by nhatthm** [comment]
Could you please update the `go.mod` files under the `tests/` folder too?

**Comment by vgarvardt** [comment]
Updated

**Pull request merged**